Menu/Tab/index.jsx

```diff
@@ -82,6 +84,16 @@ export default React.createClass({
     }
   },
 
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.resizeListener)
+  },
+
+  resizeListener () {
+    window.requestAnimationFrame
+      ? window.requestAnimationFrame(() => update(this))
```

Comment:
About the resize, yes, you absolutely should delay execution until render. The reason you are seeing the resize handler invoked so few times is because that's the maximum your computer is able to handle. If you tried it on some shitty computer from 10 years ago running Safari 7, you'd only see it invoked like twice.

And no, the updates will not be cancelled. If the resize event has managed to happen, it will throw your callback onto the event loop and invoke it the next time the stack is empty. It's the same thing as if you do something like: `setInterval(() => setTimeout(() => console.log('lol'), 0), 0)`